src/dma/buffer.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{
+    pac::dma0::channel::xfercfg::{DSTINC_A, SRCINC_A},
+    void::Void,
+};
+
+use super::{Dest, Source};
+
+impl crate::private::Sealed for &'static [u8] {}
+
+impl Source for &'static [u8] {
+    fn is_valid(&self) -> bool {
+        self.len() <= 1024
+    }
+
+    fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    fn increment(&self) -> SRCINC_A {
+        SRCINC_A::WIDTH_X_1
+    }
+
+    fn transfer_count(&self) -> Option<u16> {
```

Comment:
This isn't an issue with the internal usage, but as this trait is `pub`, a note about this method only being "valid" when it isn't empty might be nice (same for the `Dest` trait).

This would also make the comment in `start_transfer` clearer.

(Not a blocker, since it's not intended for usage by the end user)

Reply:
Good point. I pushed a commit improving the documentation of `Source`/`Dest` in general, but when it came to documenting that aspect, it just seemed so finicky. I instead changed the implementation to never underflow, at a slight performance cost.

This could be optimized by improving the design of the traits, but I don't want to get into that right now. I think it's fine to leave that for future work, whenever the need arises.

Let me know what you think.